raster-foundry/raster-foundry
# Add and update new fields annotation UI. Update import and export UI.

## Overview

This PR adds UI support for three new fields of machine generated data: `machineGenerated`, `confidence`, and `qualityCheck`. Human generated annotations using this UI will be given `false`, `null`, `null` values for these three fields, respectively. Quality check buttons are enabled.

This PR also updates import and export UI. The import UI allows you to choose properties in data to match fields recognized by the code. (E.g., the code reads `label` key and treats this as label name, but data could have other names like 'label_name' for storing such info - this updated UI handles such matching). The export UI allows you to specify a file name to save to local file system.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

<img width="1680" alt="screen shot 2017-08-23 at 10 12 04 am" src="https://user-images.githubusercontent.com/16109558/29621712-88ec0912-87ef-11e7-8f4d-5e2faa155791.png">

### Notes

Open to suggestions on the look of these new sidebar items with confidence level and quality check buttons being displayed for machine generated data.


## Testing Instructions

 * Go to `projects/edit/:projectid/annotate`
 * Upload the machine data (for example, the fake machine data below) from the new import UI and select fields to match: 
https://drive.google.com/a/azavea.com/file/d/0B6hWG0iUKKJpZlRvdW9xMVp3bFk/view?usp=sharing
<img width="1680" alt="screen shot 2017-08-23 at 10 11 11 am" src="https://user-images.githubusercontent.com/16109558/29622504-a052d8a4-87f1-11e7-8022-de5067edc11f.png">

 * Use QA check buttons to see if they work as you intended.
 * Export the QA checked data in the new export UI
<img width="1680" alt="screen shot 2017-08-23 at 10 40 35 am" src="https://user-images.githubusercontent.com/16109558/29622519-ab2cb2fe-87f1-11e7-9fbf-2775c611914e.png">

 * Upload it back to check if QA info is actually updated.
